webapp/tests/Unit/Service/ScoreboardServiceTest.php

```diff
@@ -0,0 +1,148 @@
+    {
+        self::assertEquals(
+            "99999999999999999999999.000000000",
+            ScoreboardService::convertToScoreKeyElement(0, Order::Ascending)
+        );
+
+        self::assertEquals(
+            "99999999999999999999998.000000000",
+            ScoreboardService::convertToScoreKeyElement(1, Order::Ascending)
+        );
+
+        self::assertEquals(
+            "99999999999999999999957.000000000",
+            ScoreboardService::convertToScoreKeyElement(42, Order::Ascending)
+        );
+    }
+
+    public function testScoreKeyConversionBcmath(): void
+    {
+        self::assertEquals(
+            "00000000000000000000000.000000000",
+            ScoreboardService::convertToScoreKeyElement("0")
+        );
+
+        self::assertEquals(
+            "00000000000000000000000.123000000",
+            ScoreboardService::convertToScoreKeyElement("0.123")
+        );
+
+        self::assertEquals(
+            "00000000000000000000666.123456789",
+            ScoreboardService::convertToScoreKeyElement("666.123456789")
+        );
+    }
+
+    public function testScoreKeyConversionBcmathAscending(): void
+    {
+        self::assertEquals(
+            "99999999999999999999999.000000000",
+            ScoreboardService::convertToScoreKeyElement("0", Order::Ascending)
+        );
+
+        self::assertEquals(
+            "99999999999999999999998.877000000",
+            ScoreboardService::convertToScoreKeyElement("0.123", Order::Ascending)
+        );
+
+        self::assertEquals(
+            "99999999999999999999332.876543211",
+            ScoreboardService::convertToScoreKeyElement("666.123456789", Order::Ascending)
+        );
+    }
+
+    public function testScoreKeyNegativeInt(): void
+    {
+        $this->expectException(Exception::class);
+        ScoreboardService::convertToScoreKeyElement(-1);
+    }
+
+    public function testScoreKeyNegativeBcmath(): void
+    {
+        $this->expectException(Exception::class);
+        ScoreboardService::convertToScoreKeyElement("-0.123");
+    }
+
+    public function testTooLarge(): void
+    {
+        $this->expectException(Exception::class);
+        ScoreboardService::convertToScoreKeyElement("100000000000000000000000");
+    }
+
+    public function testEmptyTeams(): void
+    {
+        $teamA = ScoreboardService::getICPCScoreKey(0, 0, 0);
+        $teamB = ScoreboardService::getICPCScoreKey(0, 0, 0);
+        self::assertEquals($teamA, $teamB);
+    }
+
+    public function testEqualTeams(): void
+    {
+        $teamA = ScoreboardService::getICPCScoreKey(7, 666, 420);
+        $teamB = ScoreboardService::getICPCScoreKey(7, 666, 420);
+        self::assertEquals($teamA, $teamB);
+    }
+
+    public function testOneTeamEmpty(): void
+    {
+        $teamA = ScoreboardService::getICPCScoreKey(0, 0, 0);
+        $teamB = ScoreboardService::getICPCScoreKey(7, 666, 420);
+        self::assertTrue($teamA < $teamB);
+    }
+
```

Comment:
```suggestion
    public function testOneTeamEmpty(): void
    {
        $teamA = ScoreboardService::getICPCScoreKey(0, 0, 0);
        $teamB = ScoreboardService::getICPCScoreKey(7, 666, 420);
        self::assertTrue($teamA < $teamB);
    }

    public function testOneTeamSolvedMore(): void
    {
        $teamA = ScoreboardService::getICPCScoreKey(1, 333, 210);
        $teamB = ScoreboardService::getICPCScoreKey(7, 666, 420);
        self::assertTrue($teamA < $teamB);
    }

```

We had all the special cases, but I think nothing to make sure that the normal case works. So the case where 1 sortcondition was higher but the others were lower was not handled yet.